src/site/resources/js/redirect.js

```diff
@@ -0,0 +1,39 @@
+window.onload = () => {
+    const currentUrl = window.location.href;
+    const urlObj = new URL(currentUrl);
+    const pathSegments = urlObj.pathname.split("/");
+    const lastSegment = pathSegments[pathSegments.length - 1];
+    const extractedUrl =  `/${lastSegment}${urlObj.hash}`;
+
+    if (currentUrl === "https://checkstyle.org/checks/") {
+        window.location.replace("https://checkstyle.org/checks");
+    }
+    else if (extractedUrl.match(/config_coding.html*/)) {
+        checkConfigCodingUrls(extractedUrl);
+    }
+    else if (extractedUrl.match(/config_header.html*/)) {
+        checkConfigHeadersUrls(extractedUrl);
+    }
+}
+
+const checkConfigCodingUrls = (url) => {
+    switch (url) {
+        case "/config_coding.html":
+            window.location.replace("https://checkstyle.org/checks/coding/");
```

Comment:
We can not hardcode `https://checkstyle.org/`, please take it from URL object

Reply:
done 

https://github.com/checkstyle/checkstyle/blob/027ff10829b98f9e3abbc997324d718e9b06cd2d/src/site/resources/js/redirect.js#L22-L29

Reply:
thanks for the suggestion! I didn't knew such dynamic routing works with `window.location.replace()` too.